### ISSUE TITLE
Shuffle in media play not working?

### ISSUE THREAD

**Issue by aimartin** [keatontaylor/alexa_media_player#203]
Hi!

I've been using alexa_media_player for quite some time and I'm really happy about how is working so far, however, I've been trying lately to use the media_play, with Spotify, and it worked, but if I try to enable shuffle mode, it doesn't work.

I've tried to use the service "media_player.shuffle_set" with the right entity_id, and passing "true", true, "True", True, 1, "1" and mane other combinations for the shuffle argument, and it doesn't work (spotify doesn't show shuffle enabled nor the following song is random if I execute a next song call)

Am I doing something wrong?

Regards

**Comment by alandtse** [comment]
It's not implemented yet.

**Comment by aimartin** [comment]
That explains a lot :D.

Any idea about when will it be included in the roadmap? (no pressure obviously, just curious :))

Regards

**Comment by alandtse** [comment]
I don't think anyone has identified the exact command the Alexa app is using to enable shuffle. We're basically reverse engineering the API.  Once that's done, we can see about implementing it.

**Comment by alandtse** [comment]
Ok, I found the command to set it and a way to read it. Probably can add it in the next release.

**Comment by aimartin** [comment]
Awesome!

Let me know if you need some help with testing or something.

Thanks a lot! :D

**Comment by alandtse** [comment]
Test version [here (use shuffle_set service)](https://github.com/alandtse/alexa_media_player/tree/shuffle).

Or play with [switches](https://github.com/alandtse/alexa_media_player/tree/switches). I'm doing more work on what switches to expose.

You should probably update to 1.3.0 before trying any test builds.

**Comment by aimartin** [comment]
Hey!

Thanks a lot for working on this and sorry, I wasn't able to test this before!

I've tested the shuffle branch version and I can confirm is working as expected :D. I'll keep this version for now and do more test with it, but so far, seems to be doing what is expected.

I'm not sure about the switches though, what should I try?

Thanks again!

**Comment by alandtse** [comment]
Thanks. Glad it works. For switches, I think I was just going to enable switches for everyone. I also added repeat and do not disturb and those make sense as switches too.

**Comment by aimartin** [comment]
I've been using the shuffle branch for a while, and I've noticed that sometimes last_used Alexa is reported wrongly (is like the variable is not updated, and is always getting the same no matter what Alexa you're talking to). If I restart HomeAssistant, Alexa works again, but after 24h (I restarted yesterday evening, and I've tried today after arriving home) is failing again.

I didn't notice it until last Saturday though, and I was using the branch version since last Monday, so maybe is not related, but never noticed it before.

Will the shuffle enhancement be merged with master?

Regards
